sync/sync-impl/src/main/java/com/duckduckgo/sync/impl/ui/qrcode/SyncBarcodeUrl.kt

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright (c) 2025 DuckDuckGo
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.duckduckgo.sync.impl.ui.qrcode
+
+import androidx.core.net.toUri
+
+data class SyncBarcodeUrl(
+    val webSafeB64EncodedCode: String,
+    val urlEncodedDeviceName: String? = null,
+) {
+
+    fun asUrl(): String {
+        val sb = StringBuilder(URL_BASE)
+            .append("&")
+            .append(CODE_PARAM).append("=").append(webSafeB64EncodedCode)
+
+        if (urlEncodedDeviceName?.isNotBlank() == true) {
+            sb.append("&")
+            sb.append(DEVICE_NAME_PARAM).append("=").append(urlEncodedDeviceName)
+        }
+
+        return sb.toString()
+    }
+
+    companion object {
+        const val URL_BASE = "https://duckduckgo.com/sync/pairing/#"
```

Comment:
this might need to change in a later PR because we’ll need to be able to access this from `app` module most likely. good for now though, just FYI it’ll probably change.